YAKL_parallel_for_c.h

```diff
@@ -594,12 +594,18 @@ namespace c {
 
   #ifdef YAKL_ARCH_SYCL
     template<class F, int N, bool simple>
-    void parallel_for_sycl( Bounds<N,simple> const &bounds , F const &f , int vectorSize ) {
-      isTriviallyCopyable<F>();
-
-      sycl_default_stream.parallel_for<class sycl_kernel>( sycl::range<1>(bounds.nIter) , [=] (sycl::id<1> i) {
-        callFunctor( f , bounds , i );
-      });
+    void parallel_for_sycl( Bounds<N,simple> const &bounds , F const &f , int vectorSize = 128 ) {
+      if constexpr (sycl::is_device_copyable<F>::value) {
+          sycl_default_stream.parallel_for( sycl::range<1>(bounds.nIter) , [=] (sycl::id<1> i) {
+              callFunctor( f , bounds , i );
+            }).wait();
+        } else {
+        F *fp = (F *) functorBuffer;
+        sycl_default_stream.memcpy(fp, &f, sizeof(F)).wait();
+        sycl_default_stream.parallel_for( sycl::range<1>(bounds.nIter) , [=] (sycl::id<1> i) {
+            callFunctor( *fp , bounds , i );
+          }).wait();
```

Comment:
Memcpy's and kernel launches were meant to be async by default. Mind if I remove `.wait()` ?

Reply:
removed `.wait()`

Reply:
For some reason, I have to keep *all* of the `.wait()` calls that you added, or the unit tests don't pass. There's something strange going on with the `sycl_default_stream` where the dependence isn't being respected fully.

Reply:
Let's address this in a separate PR, since what you have with the `.wait()` operations passes unit tests. I'll read over that article. Thanks for all the work on this!